### PR TITLE
Bump min supported swift version

### PR DESCRIPTION
**Pull request by testableapple**

### CODE REVIEW

File: .github/workflows/smoke-checks.yml [hunk 1]
```diff
@@ -223,7 +223,7 @@ jobs:
     env:
       GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       GITHUB_PR_NUM: ${{ github.event.number }}
-      XCODE_VERSION: "14.0.1"
+      XCODE_VERSION: "15.0.1"
```
Comment (laevandus) [inline .github/workflows/smoke-checks.yml:226]: 15.4?

Comment (testableapple) [reply .github/workflows/smoke-checks.yml:226]: less supported is 15.0.1 and this is the one we use in `build-xcode15` job

**Review by laevandus — APPROVED** [review]
Let's verify that Package.swift does not generate any warnings since sometimes Apple deprecates stuff in different versions.

**Comment by testableapple** [comment]
> Let's verify that Package.swift does not generate any warnings since sometimes Apple deprecates stuff in different versions.

Looks fine to me

**Pull request merged**